pkg/build/generator/generator.go

```diff
@@ -210,6 +259,19 @@ func (g *BuildGenerator) resolveImageStreamReference(ctx kapi.Context, from *kap
 
 	glog.V(4).Infof("Resolving ImageStreamReference %s of Kind %s in namespace %s", from.Name, from.Kind, namespace)
 	switch from.Kind {
+	case "ImageStream":
+		// NOTE: The 'ImageStream' reference should be used only for the 'output' image
```

Comment:
@deads2k i believe that we check this value during the BuildConfig validation (dunno if you can specify reference with Kind != ImageStream for Output.To, but I guess not).